src/single-spa-react.js

```diff
@@ -68,6 +67,15 @@ function singleSpaReact(userOpts) {
     );
   }
 
+  if (!opts.renderType) {
+    // https://github.com/single-spa/single-spa-react/issues/202
+    if (opts.ReactDOM?.createRoot || opts.ReactDOMClient?.createRoot) {
```

Comment:
createRoot is not available in react-dom < 18, so there is no need for that check.

Reply:
Good catch.

Didn't some versions before React 18 have `unstable_createRoot`? Regardless, the default should not be something unstable.

Reply:
If I remember correctly, that was during the alpha/beta versions of react 18.